linters/import/main.go

```diff
@@ -0,0 +1,438 @@
+
+	for _, decl := range f.Decls {
+		genDecl, ok := decl.(*ast.GenDecl)
+		if !ok || genDecl.Tok != token.IMPORT {
+			break
+		}
+
+		var (
+			importDecl importDecl
+			group      importGroup
+		)
+
+		var lastLine int
+		for _, spec := range genDecl.Specs {
+			importSpec := spec.(*ast.ImportSpec)
+			pos := importSpec.Path.ValuePos
+			line := fset.Position(pos).Line
+			fileName := fset.Position(pos).Filename
+			if lastLine > 0 && pos > 0 && line-lastLine > 1 {
+				importDecl.Groups = append(importDecl.Groups, group)
+				group = importGroup{}
+			}
+			group.Imports = append(group.Imports, newImportSpec(importSpec, line, fileName))
+			lastLine = line
+		}
+		importDecl.Groups = append(importDecl.Groups, group)
+		importDecls = append(importDecls, importDecl)
+	}
+
+	return importDecls
+}
+
+func filterOutVendor(importPaths []string) []string {
+	filteredStrings := []string{}
+	for _, importPath := range importPaths {
+		if !strings.Contains(importPath, "/vendor/") {
+			filteredStrings = append(filteredStrings, importPath)
+		}
+	}
+	return filteredStrings
+}
+
+func newImportSpec(is *ast.ImportSpec, line int, filename string) importSpec {
+	var (
+		path string
+	)
+
+	pathLit := is.Path
+	if pathLit != nil {
+		path = pathLit.Value
+	}
+
+	return importSpec{
+		Name: filename,
+		Path: path,
+		Line: line,
+	}
+}
+
+func findErrors(importDecls []importDecl, patterns []string, file *ast.File) lintErrors {
+	var (
+		lintErrors lintErrors
+		// checkedPatterns is used to keep track of patterns we've seen, but want to use again
+		checkedPatterns []string
+
+		// seenPatterns is used to keep track of patterns that have been seen already
+		seenPatterns = make(map[string]interface{})
+	)
+
+	// check if there is more than one import group
+	if len(importDecls) > 1 {
+		lintErr := lintError{
+			fileName: importDecls[0].Groups[0].Imports[0].Name,
+			err:      errMultipleImport,
+		}
+		lintErrors = append(lintErrors, lintErr)
+		return lintErrors
+	}
+
+	return findRecErr(importDecls[0].Groups, checkedPatterns, patterns, lintErrors, seenPatterns)
+}
+
+func findRecErr(importGroup []importGroup, checkedPatterns, currentPatterns []string, lintErrors lintErrors, seenPatterns map[string]interface{}) lintErrors {
+	for _, ind := range importGroup {
+		checkedPatterns = checkedPatterns[:0]
+		var checking bool
+		for idx, i := range ind.Imports {
+			if len(currentPatterns) == 0 && len(importGroup) > 0 && len(checkedPatterns) > 0 {
+				lintErrors = addLintError(i, lintErrors, errNoMatch, "")
+
+				empty := checkedPatterns[:0] // clear out checkedPatterns before moving on to groups (could probably get rid of this)
+				return findRecErr(importGroup[1:], empty, checkedPatterns, lintErrors, seenPatterns)
+			}
+			if len(currentPatterns) == 0 && len(importGroup) > 0 {
+				return addLintError(i, lintErrors, errTooManyImports, "")
+			}
+			if currentPatterns[0] == standard {
+				match, err := regexp.MatchString(`\.`, i.Path)
```

Comment:
this snippet of code is repeated a few different times, could you extract into a function:

```
func isThirdParty(path string) bool {
  re := regex.MustCompile(`\.`)
  return re.MatchString(path)
}
```